StudyHub/StudyHub.BLL/Services/SubjectService.cs

```diff
@@ -121,4 +95,66 @@ public async Task<SubjectDTO> UpdateSubjectAsync(Guid userId, Guid subjectId, Up
 
         return _mapper.Map<SubjectDTO>(entity);
     }
+
+    public async Task<List<StudentDTO>> AddStudentsToSubjectAsync(
+        Guid subjectId,
+        Guid teacherId,
+        StudentsToSubjectRequest request)
+    {
+        var subject = await _subjectRepository.FirstOrDefaultAsync(s => s.Id == subjectId)
+            ?? throw new NotFoundException("Subject not found with the specified Id");
+
+        if (subject.TeacherId != teacherId)
+            throw new RestrictedAccessException("You are not the owner and do not have permission to perform this action.");
+
+        foreach (var email in request.Emails)
+        {
+            var user = await _userManager.FindByEmailAsync(email)
+                ?? throw new NotFoundException($"Student not found with the specified email: {email}");
+
+            user.Subjects ??= new List<Subject>();
```

Comment:
not really get why you create SubjectList. I got, that you not inlude navigation prop Subject(idk if we can do it with userManager), but I wanted to ask, if it working correctly in that case if we already got some subjects in user. I think, that it delete(probably) all current subject and put only one that you got here. Double check it please

Reply:
it not delete all current subject because i use ??=